File: bindings/javascript/Polycode/ProgramResourceLoader.js
function ProgramResourceLoader() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.ProgramResourceLoader()
	}
}


Duktape.fin(ProgramResourceLoader.prototype, function (x) {
	if (x === ProgramResourceLoader.prototype) {
		return;
	}
	Polycode.ProgramResourceLoader__delete(x.__ptr)
})

ProgramResourceLoader.prototype.loadResource = function(path,targetPool) {
	var retVal = new Resource()
	retVal.__ptr = Polycode.ProgramResourceLoader_loadResource(this.__ptr, path, targetPool.__ptr)
	return retVal
}

File: bindings/lua/Polycode/ProgramResourceLoader.lua
class "ProgramResourceLoader"


function ProgramResourceLoader:ProgramResourceLoader(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.ProgramResourceLoader(unpack(arg))
	end
end

function ProgramResourceLoader:__delete()
	if self then Polycode.delete_ProgramResourceLoader(self.__ptr) end
end
